src/libraries/System.Private.CoreLib/src/System/Numerics/BFloat16.cs

```diff
@@ -0,0 +1,1984 @@
+        /// <exception cref="ArgumentException">Thrown when <paramref name="obj"/> is not of type <see cref="BFloat16"/>.</exception>
+        public int CompareTo(object? obj)
+        {
+            if (obj is not BFloat16 other)
+            {
+                return (obj is null) ? 1 : throw new ArgumentException(SR.Arg_MustBeBFloat16);
+            }
+            return CompareTo(other);
+        }
+
+        /// <summary>
+        /// Compares this object to another object, returning an integer that indicates the relationship.
+        /// </summary>
+        /// <returns>A value less than zero if this is less than <paramref name="other"/>, zero if this is equal to <paramref name="other"/>, or a value greater than zero if this is greater than <paramref name="other"/>.</returns>
+        public int CompareTo(BFloat16 other) => ((float)this).CompareTo((float)other);
+
+        /// <inheritdoc cref="IEqualityOperators{TSelf, TOther, TResult}.op_Equality(TSelf, TOther)" />
+        public static bool operator ==(BFloat16 left, BFloat16 right) => (float)left == (float)right;
+
+        /// <inheritdoc cref="IEqualityOperators{TSelf, TOther, TResult}.op_Inequality(TSelf, TOther)" />
+        public static bool operator !=(BFloat16 left, BFloat16 right) => (float)left != (float)right;
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_LessThan(TSelf, TOther)" />
+        public static bool operator <(BFloat16 left, BFloat16 right) => (float)left < (float)right;
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_GreaterThan(TSelf, TOther)" />
+        public static bool operator >(BFloat16 left, BFloat16 right) => (float)left > (float)right;
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_LessThanOrEqual(TSelf, TOther)" />
+        public static bool operator <=(BFloat16 left, BFloat16 right) => (float)left <= (float)right;
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_GreaterThanOrEqual(TSelf, TOther)" />
+        public static bool operator >=(BFloat16 left, BFloat16 right) => (float)left >= (float)right;
+
+        // Equality
+
+        /// <summary>
+        /// Returns a value that indicates whether this instance is equal to a specified <paramref name="other"/> value.
+        /// </summary>
+        public bool Equals(BFloat16 other) => ((float)this).Equals((float)other);
+
+        /// <summary>
+        /// Returns a value that indicates whether this instance is equal to a specified <paramref name="obj"/>.
+        /// </summary>
+        public override bool Equals(object? obj) => obj is BFloat16 other && Equals(other);
+
+        /// <summary>
+        /// Serves as the default hash function.
+        /// </summary>
+        public override int GetHashCode() => ((float)this).GetHashCode();
```

Comment:
I'd imagine this one is also cheaper to implement in software

Reply:
Did some simple test of operator <=:

| Method         | left          | right       | Mean      | Error     | StdDev    | Median    | Ratio | RatioSD |
|--------------- |-------------- |------------ |----------:|----------:|----------:|----------:|------:|--------:|
| IntegerBitMath | 0             | 0           | 0.4893 ns | 0.0050 ns | 0.0044 ns | 0.4884 ns |  5.18 |    0.28 |
| ConvertToFloat | 0             | 0           | 0.0946 ns | 0.0056 ns | 0.0050 ns | 0.0936 ns |  1.00 |    0.00 |
|                |               |             |           |           |           |           |       |         |
| IntegerBitMath | 4.9370763E-36 | 9.1835E-41  | 0.4608 ns | 0.0071 ns | 0.0063 ns | 0.4618 ns |     ? |       ? |
| ConvertToFloat | 4.9370763E-36 | 9.1835E-41  | 0.0058 ns | 0.0078 ns | 0.0073 ns | 0.0000 ns |     ? |       ? |
|                |               |             |           |           |           |           |       |         |
| IntegerBitMath | 9.1835E-41    | -9.1835E-41 | 0.4623 ns | 0.0069 ns | 0.0065 ns | 0.4625 ns |     ? |       ? |
| ConvertToFloat | 9.1835E-41    | -9.1835E-41 | 0.0182 ns | 0.0096 ns | 0.0170 ns | 0.0143 ns |     ? |       ? |
|                |               |             |           |           |           |           |       |         |
| IntegerBitMath | NaN           | 0           | 0.2463 ns | 0.0055 ns | 0.0052 ns | 0.2455 ns | 20.29 |    9.85 |
| ConvertToFloat | NaN           | 0           | 0.0149 ns | 0.0071 ns | 0.0066 ns | 0.0158 ns |  1.00 |    0.00 |
|                |               |             |           |           |           |           |       |         |
| IntegerBitMath | ∞             | 0           | 0.4565 ns | 0.0072 ns | 0.0064 ns | 0.4539 ns | 44.46 |   16.61 |
| ConvertToFloat | ∞             | 0           | 0.0120 ns | 0.0043 ns | 0.0040 ns | 0.0121 ns |  1.00 |    0.00 |

And no inline version, the result is extremely unstable:

| Method                 | left          | right       | Mean      | Error     | StdDev    | Ratio | RatioSD |
|----------------------- |-------------- |------------ |----------:|----------:|----------:|------:|--------:|
| IntegerBitMathNoInline | 0             | 0           | 2.9355 ns | 0.0802 ns | 0.1014 ns |  3.79 |    0.88 |
| ConvertToFloatNoInline | 0             | 0           | 0.8504 ns | 0.0576 ns | 0.1661 ns |  1.00 |    0.00 |
|                        |               |             |           |           |           |       |         |
| IntegerBitMathNoInline | 4.9370763E-36 | 9.1835E-41  | 3.0532 ns | 0.1256 ns | 0.3625 ns |  4.66 |    1.75 |
| ConvertToFloatNoInline | 4.9370763E-36 | 9.1835E-41  | 0.7179 ns | 0.0696 ns | 0.1975 ns |  1.00 |    0.00 |
|                        |               |             |           |           |           |       |         |
| IntegerBitMathNoInline | 9.1835E-41    | -9.1835E-41 | 2.9630 ns | 0.0803 ns | 0.1341 ns |  3.52 |    0.91 |
| ConvertToFloatNoInline | 9.1835E-41    | -9.1835E-41 | 0.8768 ns | 0.0547 ns | 0.1596 ns |  1.00 |    0.00 |
|                        |               |             |           |           |           |       |         |
| IntegerBitMathNoInline | NaN           | 0           | 1.8358 ns | 0.0766 ns | 0.2187 ns |  2.22 |    0.68 |
| ConvertToFloatNoInline | NaN           | 0           | 0.8883 ns | 0.0891 ns | 0.2528 ns |  1.00 |    0.00 |
|                        |               |             |           |           |           |       |         |
| IntegerBitMathNoInline | ∞             | 0           | 2.7604 ns | 0.0702 ns | 0.1301 ns |  3.23 |    0.67 |
| ConvertToFloatNoInline | ∞             | 0           | 0.9039 ns | 0.0711 ns | 0.2028 ns |  1.00 |    0.00 |